subprojects/ide/src/main/java/org/gradle/plugins/ide/api/PropertyListGeneratorTask.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.plugins.ide.api;
```

Comment:
I modeled this after `XmlGeneratorTask` and `PropertiesGeneratorTask`. I figure we may offer the same advantage as those two tasks. It's also useful for generating `.xcsettings` and `.pbxproj`. A user can access the low-level generator and use it as a last chance to transform the file however they please before been written to disk. It's quite useful from my experience with Visual Studio.